crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -1266,29 +1292,47 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
     /// After updating the timeline item `new_item` which id is
     /// `target_event_id`, update other items that are responses to this item.
     fn maybe_update_responses(
+        meta: &mut TimelineMetadata,
         items: &mut ObservableItemsTransaction<'_>,
         target_event_id: &EventId,
         new_item: &EventTimelineItem,
     ) {
-        items.for_each(|mut entry| {
-            let Some(event_item) = entry.as_event() else { return };
-            let Some(message) = event_item.content.as_message() else { return };
-            let Some(in_reply_to) = message.in_reply_to() else { return };
-            if target_event_id == in_reply_to.event_id {
-                trace!(reply_event_id = ?event_item.identifier(), "Updating response to edited event");
-                let in_reply_to = InReplyToDetails {
-                    event_id: in_reply_to.event_id.clone(),
-                    event: TimelineDetails::Ready(Box::new(
-                        RepliedToEvent::from_timeline_item(new_item),
-                    )),
-                };
-                let new_reply_content =
+        let Some(replies) = meta.replies.get(target_event_id) else {
+            trace!("item has no replies");
+            return;
+        };
+
+        for reply_id in replies {
+            let Some(timeline_item_index) = items
+                .get_remote_event_by_event_id(reply_id)
+                .and_then(|meta| meta.timeline_item_index)
+            else {
+                warn!(%reply_id, "event not known as a timeline item in the timeline");
```

Comment:
nit: 'as an item in the timeline' or 'as a timeline item in this timeline' maybe?